policy-controller/k8s/index/src/index.rs

```diff
@@ -1172,11 +1176,71 @@ impl PolicyIndex {
         authzs
     }
 
-    fn http_routes(&self, server_name: &str) -> HashMap<String, InboundHttpRoute> {
+    fn route_client_authzs(
+        &self,
+        route_name: &str,
+        authentications: &AuthenticationNsIndex,
+    ) -> HashMap<AuthorizationRef, ClientAuthorization> {
+        let mut authzs = HashMap::default();
+
+        for (name, spec) in self.authorization_policies.iter() {
+            // Skip the policy if it doesn't apply to the route.
+            if let authorization_policy::Target::HttpRoute(name) = &spec.target {
+                if name != route_name {
+                    tracing::trace!(
+                        ns = %self.namespace,
+                        authorizationpolicy = %name,
+                        route = %route_name,
+                        target = %name,
+                        "AuthorizationPolicy does not target HttpRoute",
+                    );
+                    continue;
+                }
+            } else {
+                continue;
+            }
```

Comment:
the control flow here feels a _little_ hard to follow, with the multiple `continue`s...i'm not really sure if there is a way to make it much simpler, so i'm fine with it as-is, but i don't _love_ it...

Reply:
(side note, in Rust 1.64, we'll be able to use the `if let` chains that stabilized in https://github.com/rust-lang/rust/pull/94927#event-7007028976, which might simplify this a little)

Reply:
probably clearer with a match:

```suggestion
            match &spec.target {
                authorization_policy::Target::HttpRoute(n) if n == route_name => {}
                _ => {
                    tracing::trace!(
                        ns = %self.namespace,
                        authorizationpolicy = %name,
                        route = %route_name,
                        target = %name,
                        "AuthorizationPolicy does not target HttpRoute",
                    );
                    continue;
                }
            }
```